enhancements/network/upi-node-ip-selection.md

```diff
@@ -0,0 +1,216 @@
+---
+title: upi-node-ip-selection
+authors:
+  - "@pliu"
+reviewers: 
+  - "@racedo"
+  - "@makentenza"
+  - "@cybertron"
+  - "@mkowalski"
+approvers:
+  - "@danwinship"
+  - "@trozet"
+api-approvers: 
+  - "None"
+creation-date: 2023-02-08
+last-updated: 2023-02-08
+tracking-link:
+  - https://issues.redhat.com/browse/NP-687
+see-also:
+  - "/enhancements/network/ip-interface-selection"
+replaces:
+superseded-by:
+---
+
+# Node IP Selection UPI
+
+## Summary
+
+When Openshift is deployed on user-provisioned infrastructure, and there are
+more than one NIC available on the hosts. We need a reliable way to guarantee
+all the openshift components can use the correct IP/interface.
+
+## Motivation
+
+Today, on all the UPI platforms (AWS, GCP, vSphere, IBM Z and LinuxONE with
```

Comment:
This is not true. nodeip-configuration is enabled on all platform None deployments: https://github.com/openshift/machine-config-operator/blob/master/templates/common/_base/units/nodeip-configuration.service.yaml#L2

Notably, that excludes VSphere UPI because it sets platform to VSphere and not None, but all of the other platforms should have this enabled already. VSphere should have it too once I finally get https://github.com/openshift/machine-config-operator/pull/3460 working.

Is it possible you meant IPI here? We only enable nodeip-configuration for on-prem IPI (baremetal, vsphere, openstack, etc.) so many of these platforms will indeed not have nodeip-configuration in IPI deployments.

Reply:
Actually, this also confuses me. What does the platform None exactly mean? In OCP doc, I can only find the platform is set to 'none' for UPI bare-metal and IBM platforms. Do we have nodeip-configuration.service enabled for UPI on cloud platforms?

Currently, this EPIC only targets UPI platforms. But I'm wondering whether we want to have the IPI platforms also use the same approach.

Reply:
Ah, this is a terminology problem. UPI is not necessarily baremetal, but unfortunately sometime in the past "baremetal" was used as a generic way to describe UPI deployments, despite being technically incorrect in a lot of cases. It is possible to do UPI deployments on baremetal, but not all UPI deployments are baremetal.

To avoid confusion, I would suggest only using "baremetal" to describe the platform where OpenShift is installed directly to the hardware with no virt layer in between. This may be either IPI or UPI. "UPI" is the deployment mechanism where platform is set to None (except the aforementioned exception for VSphere) and the installer is not responsible for deploying the infrastructure.

There are actually separate conversations going on right now to fix the incorrect use of baremetal in our docs, but unfortunately nothing new has been published yet.